ftdetect/yagpdbcc.vim

```diff
@@ -34,10 +37,12 @@ au BufRead,BufNewFile   *.yag-cc      setfiletype yagpdbcc
 au BufRead,BufNewFile   *.yagpdbcc    setfiletype yagpdbcc
 au BufRead,BufNewFile   *.yagpdb-cc   setfiletype yagpdbcc
 
-" Also use *.tmpl, *.gotmpl et al., which are originally only Go.
-if get(g:, 'yagpdbcc_override_ft')
-    " Here, we need to explicitly override the default "template" syntax for
-    " .tmpl files:
-    au BufRead,BufNewFile   *.tmpl    setlocal filetype=yagpdbcc
-    au BufRead,BufNewFile   *.gotmpl  setfiletype yagpdbcc
-endif
+" Also use *.tmpl, *.gotmpl et al., which are originally only Go, if configured.
+" Here, we need to explicitly override the default syntax - "setfiletype"
+" will not override an existing filetype.
+au BufRead,BufNewFile   *.tmpl,*.gotmpl
+    \ if yagpdbcc#config#OverrideFt() | setlocal filetype=yagpdbcc | endif
```

Comment:
I don't want to step on your toes here — you originally moved these to `plugin/yagpdbcc.vim`, but I think they fit better here, with the other filetype autocommands. Is this all right?

Reply:
I was worried about ftplugin being sourced too early to not obey the user's configuration, no matter where it popped up, so I used plugin instead.

If it works in ftplugin, keep it there.